kite-data/kite-data-oozie/src/test/java/org/kitesdk/data/oozie/TestKiteURIHandler.java

```diff
@@ -215,6 +238,90 @@ public void existsForNonReadiableView() throws URIHandlerException, URISyntaxExc
+    uriHandler.exists(uri, null);
+
+    defaultConf = DefaultConfiguration.get();
+    Assert.assertEquals("test.value", defaultConf.get("test.property"));
+    Assert.assertEquals("something.else", Services.get().get(HCatAccessorService.class).getHCatConf().get("test.value"));
+  }
+
+  @Test
+  public void noConfigLoadedWhenNoServices() throws URIHandlerException, URISyntaxException {
+    URI uri = new URI("view:file:target/data/data/nomailbox?message=hello");
+    uriHandler.exists(uri, null);
+    Assert.assertNull(Services.get());
+  }
+
+  @Test
+  public void noConfigLoadedWhenNoHCatAccessor() throws URIHandlerException, URISyntaxException, ServiceException, FileNotFoundException, IOException {
+    setupHCatService(false, false);
+
+    URI uri = new URI("view:file:target/data/data/nomailbox?message=hello");
+    uriHandler.exists(uri, null);
+    Assert.assertNotNull(Services.get());
+    Assert.assertNull(Services.get().get(HCatAccessorService.class));
+  }
+
+  @Test
+  public void noConfigLoadedIfNoHCatAccessorConfig() throws URIHandlerException, URISyntaxException, ServiceException, FileNotFoundException, IOException {
+    setupHCatService(true, false);
+    URI uri = new URI("view:file:target/data/data/nomailbox?message=hello");
+    uriHandler.exists(uri, null);
+    Assert.assertNotNull(Services.get());
+    Assert.assertNotNull(Services.get().get(HCatAccessorService.class));
+    Assert.assertNull(Services.get().get(HCatAccessorService.class).getHCatConf());
+  }
+
+  private void setupHCatService(boolean loadHCat, boolean loadHCatConfig) throws ServiceException, FileNotFoundException, IOException {
+    serviceTempDir = Files.createTempDir();
+    File confDir = new File(serviceTempDir, "conf");
+    File hadoopConfDir = new File(confDir, "hadoop-conf");
+    File hadoopConfTarget = new File(hadoopConfDir, "hadoop-site.xml");
+    File actionConfDir = new File(confDir, "action-conf");
+    File hiveConfDir = new File(confDir, "hive-conf");
+    File hiveConfTarget = new File(hiveConfDir, "hive-site.xml");
+    File oozieSiteTarget = new File(confDir, "oozie-site.xml");
+    confDir.mkdir();
+    hadoopConfDir.mkdir();
+    actionConfDir.mkdir();
+    hiveConfDir.mkdir();
+
+    Configuration oozieSiteConf = new Configuration(false);
+
+    if(loadHCatConfig) {
+      oozieSiteConf.set("oozie.service.HCatAccessorService.hcat.configuration", hiveConfTarget.getAbsolutePath());
+    }
+
+    oozieSiteConf.set("oozie.services", "org.apache.oozie.service.HadoopAccessorService");
+    oozieSiteConf.writeXml(new FileOutputStream(oozieSiteTarget));
+    conf.writeXml(new FileOutputStream(hadoopConfTarget));
+
+    Configuration hiveConf = new Configuration(false);
+    hiveConf.set("test.property", "test.value");
+    hiveConf.writeXml(new FileOutputStream(hiveConfTarget));
+
+    // set to the temp directory
+    System.setProperty("oozie.home.dir", serviceTempDir.getAbsolutePath());
+
+    Services services = new Services();
```

Comment:
We need to make sure to destroy the Services object after each test.  We used to have a lot of flakey tests in Oozie because we were missing some of these, so Services would bleed into other tests causing all kinds of weird and hard-to-diagnose problems.

We typically take care of this by creating it in the setUp() method and destroying in the tearDown() method.  
```
private Services services;
@Override
protected void setUp() throws Exception {
   super.setUp();
    services = new Services();
    services.init();
}
@Override
protected void tearDown() throws Exception {
    services.destroy();
    super.tearDown();
}
```
For tests that require more flexibility or creating/destroying the Services, we usually use a try-finally block:
```
Services services = null;
try {
   services = new Services();
   services.init();
   // do stuff
} finally {
   if (services != null) {
      services.destroy();
   }
}
```